tests/test_main.py

```diff
@@ -20,3 +20,49 @@ def test_unknown_raises(fa_markdown):
     with pytest.raises(FontAwesomeException) as exc:
         fa_markdown.convert("i ♥ :fa-{0}:".format(unknown_icon))
         assert unknown_icon in exc.message
+
+
+def test_size(fa_markdown):
+    expected_markup = '<p>i ♥ <i class="fa fa-coffee fa-xs"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-coffee fa-xs:') == expected_markup
+
+    expected_markup = '<p>i ♥ <i class="fa fa-coffee fa-sm"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-coffee fa-sm:') == expected_markup
+
+    expected_markup = '<p>i ♥ <i class="fa fa-coffee fa-lg"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-coffee fa-lg:') == expected_markup
+
+    expected_markup = '<p>i ♥ <i class="fa fa-coffee fa-1x"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-coffee fa-1x:') == expected_markup
+
+    expected_markup = '<p>i ♥ <i class="fa fa-coffee fa-3x"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-coffee fa-3x:') == expected_markup
+
+    expected_markup = '<p>i ♥ <i class="fa fa-coffee fa-10x"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-coffee fa-10x:') == expected_markup
+
+
+def test_fab_icon(fa_markdown):
+    expected_markup = '<p>i ♥ <i class="fab fa-facebook"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-facebook:') == expected_markup
+
+
+def test_fab_icon_with_prefix(fa_markdown):
+    expected_markup = '<p>i ♥ <i class="fab fa-facebook"></i></p>'
+    assert fa_markdown.convert('i ♥ :fab fa-facebook:') == expected_markup
+
+
+def test_far_icon_with_prefix(fa_markdown):
+    expected_markup = '<p>i ♥ <i class="far fa-font-awesome-logo-full"></i></p>'
+    assert fa_markdown.convert('i ♥ :far fa-font-awesome-logo-full:') == expected_markup
+
+
+def test_fa_icon_without_prefix(fa_markdown):
+    expected_markup = '<p>i ♥ <i class="fa fa-font-awesome-logo-full"></i></p>'
+    assert fa_markdown.convert('i ♥ :fa-font-awesome-logo-full:') == expected_markup
+
+
+# light style not found in icons.json...
+# def test_fal_icon_with_prefix(fa_markdown):
+    # expected_markup = '<p>i ♥ <i class="fal fa-font-awesome-logo-full"></i></p>'
+    # assert fa_markdown.convert('i ♥ :fal fa-address-book:') == expected_markup
```

Comment:
What's the problem here?

Reply:
some icons have multi prefixes. eg. fa-address-book has 'solid', 'regular' and 'light' prefixes.

https://fontawesome.com/icons/address-book?style=solid

but I can't find light prefix in [metadata](https://github.com/FortAwesome/Font-Awesome/blob/master/advanced-options/metadata/icons.json) 

```json
  "address-book": {
    "changes": [
      "4.7",
      "5.0.0",
      "5.0.3"
    ],
    "ligatures": [],
    "search": {
      "terms": []
    },
    "styles": [
      "solid",
      "regular"
    ],
    "unicode": "f2b9",
    "label": "Address Book",
    "svg": {
      "solid": {
        "last_modified": 1525209365064,
        "raw": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 448 512\"><path d=\"M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z\"/></svg>",
        "viewBox": [
          "0",
          "0",
          "448",
          "512"
        ],
        "width": 448,
        "height": 512,
        "path": "M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
      },
      "regular": {
        "last_modified": 1525209365040,
        "raw": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 448 512\"><path d=\"M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-68 304H48V48h320v416zM208 256c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm-89.6 128h179.2c12.4 0 22.4-8.6 22.4-19.2v-19.2c0-31.8-30.1-57.6-67.2-57.6-10.8 0-18.7 8-44.8 8-26.9 0-33.4-8-44.8-8-37.1 0-67.2 25.8-67.2 57.6v19.2c0 10.6 10 19.2 22.4 19.2z\"/></svg>",
        "viewBox": [
          "0",
          "0",
          "448",
          "512"
        ],
        "width": 448,
        "height": 512,
        "path": "M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-68 304H48V48h320v416zM208 256c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm-89.6 128h179.2c12.4 0 22.4-8.6 22.4-19.2v-19.2c0-31.8-30.1-57.6-67.2-57.6-10.8 0-18.7 8-44.8 8-26.9 0-33.4-8-44.8-8-37.1 0-67.2 25.8-67.2 57.6v19.2c0 10.6 10 19.2 22.4 19.2z"
      }
    }
  }
```

I think this [metadata](https://github.com/FortAwesome/Font-Awesome/blob/master/advanced-options/metadata/icons.json) might include only free icons.

Reply:
I wander if I should skip prefix check?